File: modules/sql/app/model/conditioncorrector/ExtractedExpressions.java
package model.conditioncorrector;

import java.util.List;

import net.sf.jsqlparser.expression.BinaryExpression;
import net.sf.jsqlparser.expression.Expression;

public class ExtractedExpressions {
  
  private List<BinaryExpression> binaryExpressions;

  private List<Expression> otherExpressions;

  public ExtractedExpressions(List<BinaryExpression> theBinaryExpressions, List<Expression> theOtherExpressions) {
    binaryExpressions = theBinaryExpressions;
    otherExpressions = theOtherExpressions;
  }

  public List<BinaryExpression> getBinaryExpressions() {
    return binaryExpressions;
  }

  public List<Expression> getOtherExpressions() {
    return otherExpressions;
  }

  public boolean isEmpty() {
    return binaryExpressions.isEmpty() && otherExpressions.isEmpty();
  }
  
}

File: modules/sql/app/model/conditioncorrector/ExtractedExpressions.scala
package model.conditioncorrector;

import net.sf.jsqlparser.expression.BinaryExpression
import net.sf.jsqlparser.expression.Expression

class ExtractedExpressions(binaryExpressions: List[BinaryExpression], otherExpressions: List[Expression]) {

  def isEmpty() = binaryExpressions.isEmpty && otherExpressions.isEmpty

}
